### PR TITLE
Fix rustc and clippy warnings

### PR DESCRIPTION
**Pull request by DaniPopes**
## Description

Fix rustc and clippy warnings, mostly automated with `cargo clippy --fix`.

## How This Has Been Tested

`cargo clippy --all-targets` on the latest few LLVM versions.

## Checklist

- [x] I have read the [Contributing Guide](https://github.com/TheDan64/inkwell/blob/master/.github/CONTRIBUTING.md)

### CODE REVIEW

**Comment by TheDan64** [comment]
Please update the PR and fix broken tests

File: src/context.rs [hunk 1]
```diff
@@ -339,6 +341,7 @@ impl ContextImpl {
         }
     }
 
+    #[allow(deprecated)]
     fn metadata_string<'ctx>(&self, string: &str) -> MetadataValue<'ctx> {
```
Comment (TheDan64) [inline src/context.rs:345]: What are these replaced by?

File: src/support/mod.rs [hunk 2]
```diff
@@ -213,7 +214,7 @@ pub fn enable_llvm_pretty_stack_trace() {
 /// A) Finds a terminating null byte in the Rust string and can reference it directly like a C string.
 ///
 /// B) Finds no null byte and allocates a new C string based on the input Rust string.
-pub(crate) fn to_c_str<'s>(mut s: &'s str) -> Cow<'s, CStr> {
+pub(crate) fn to_c_str(mut s: &str) -> Cow<'_, CStr> {
```
Comment (TheDan64) [inline src/support/mod.rs:217]: Did this trigger a clippy warning? Would prefer to be explicit about the lifetimes here if not

File: src/types/int_type.rs [hunk 3]
```diff
@@ -58,14 +57,7 @@ impl StringRadix {
...
         // and all digits must be in the radix' character set
-        let mut it = slice.chars();
-        match self {
-            StringRadix::Binary => it.all(|c| matches!(c, '0'..='1')),
-            StringRadix::Octal => it.all(|c| matches!(c, '0'..='7')),
-            StringRadix::Decimal => it.all(|c| matches!(c, '0'..='9')),
-            StringRadix::Hexadecimal => it.all(|c| matches!(c, '0'..='9' | 'a'..='f' | 'A'..='F')),
-            StringRadix::Alphanumeric => it.all(|c| matches!(c, '0'..='9' | 'a'..='z' | 'A'..='Z')),
-        }
+        slice.chars().all(|c| c.is_digit(*self as u32))
```
Comment (TheDan64) [inline src/types/int_type.rs:60]: I don't think this is correct since hex and alpha can have numerics

**Review by TheDan64 — CHANGES_REQUESTED** [review]

File: src/values/instruction_value.rs [hunk 4]
```diff
@@ -840,14 +846,6 @@ impl<'ctx> InstructionValue<'ctx> {
     }
 }
 
-impl Clone for InstructionValue<'_> {
-    /// Creates a clone of this `InstructionValue`, and returns it.
-    /// The clone will have no parent, and no name.
-    fn clone(&self) -> Self {
-        unsafe { InstructionValue::new(LLVMInstructionClone(self.as_value_ref())) }
```
Comment (TheDan64) [inline src/values/instruction_value.rs:847]: We should note this in the next release as a breaking change..

File: src/support/mod.rs (hunk 2 above)
Comment (DaniPopes) [reply src/support/mod.rs:217]: Yes, I believe it's https://rust-lang.github.io/rust-clippy/master/index.html#/needless_lifetimes

File: src/values/instruction_value.rs (hunk 4 above)
Comment (TheDan64) [reply src/values/instruction_value.rs:847]: The current behavior seems expected, no? Sort of like cloning a String, you get a totally different allocation?

Comment (DaniPopes) [reply src/values/instruction_value.rs:847]: The problem is it also has a Copy impl, which has different behavior to the Clone one.
https://rust-lang.github.io/rust-clippy/master/index.html#/non_canonical_clone_impl

Copy docs also say:
> If a type is Copy then its [Clone](https://doc.rust-lang.org/stable/std/clone/trait.Clone.html) implementation only needs to return *self (see the example above).

https://doc.rust-lang.org/stable/std/marker/trait.Copy.html#whats-the-difference-between-copy-and-clone

File: src/types/int_type.rs (hunk 3 above)
Comment (TheDan64) [reply src/types/int_type.rs:60]: Oh interesting, I didn't know that. But what about subsets? For example, binary should only be 0 or 1, and not 2-9, a-z, A-Z, etc?

Comment (DaniPopes) [reply src/types/int_type.rs:60]: Yes, that's what the radix argument is for :)

Comment (TheDan64) [reply src/types/int_type.rs:60]: ohhhh haha

**Review by TheDan64 — APPROVED** [review]
Thanks!

**Pull request merged**